lib/manifest/EndListAppender.js

```diff
@@ -57,12 +58,14 @@ var EndlistAppender = function EndListAppender(){
                 return;
             }
             var timeoutObj = setTimeout(function () {
-                appendEndlistIfMissing.call(that, entryId).then(function(){
-                    return that.removeEntry(entryId);
-                }).catch(function(err){
-                    logger.error('Error adding EXT-X-ENDLIST: ' + err.message);
-                });
-            }, sessionManager.getSessionDuration());
+                logger.debug("Timer ended. Starting endlistAppender");
+                appendEndlistIfMissing.call(that, entryId)
+                    .then(function(){
+                        return that.removeEntry(entryId);
+                    }).catch(function(err){
+                        logger.error('Error adding EXT-X-ENDLIST: ' + err.message);
```

Comment:
logger.error('Error adding EXT-X-ENDLIST:', ErrorUtils.error2string(err));
And of course  add 
 var ErrorUtils = require('./../utils/error-utils');